pkg/api/types.go

```diff
@@ -200,6 +200,8 @@ type VolumeSource struct {
 	ISCSI *ISCSIVolumeSource `json:"iscsi"`
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
+	// CinderPersistentDisk represents a cinder volume attached and mounted on kubelets host machine
+	CinderPersistentDisk *CinderPersistentDiskVolumeSource `json:"cinderPersistentDisk"`
```

Comment:
You'll also want this in the PersistentVolumeSource struct if Cinder is to be provisioned as a PersistentVolume.